### PR TITLE
Pattern fill and stroke

### PR DESCRIPTION
**Pull request by mcm69**

### CODE REVIEW

**Comment by mcm69** [comment]
@skavish @shamansir please review

File: src/anm/graphics/brush.js [hunk 1]
```diff
@@ -176,6 +176,21 @@ Brush.prototype.adapt = function(ctx) {
...
+            fill;
+        var canvas = engine.createCanvas(this.pattern.w, this.pattern.h, null, 1);
+        var cctx = canvas.getContext('2d');
+        elm.pivot(0,0);
+        elm.disabled = false;
+        elm.render(cctx, 0, 0);
+        elm.disabled = true;
+        fill = canvas;
+
+        if (fill) {
```
Comment (shamansir) [inline src/anm/graphics/brush.js]: Why `fill` could be `false` here?

File: src/import/animatron-importer.js [hunk 2]
```diff
@@ -744,8 +748,16 @@ Import.fill = function(src) {
...
+    var fill;
+    if (is.str(src[0])) {
+        fill = src[0];
+    } else if (is.arr(src[0])) {
+        if (is.arr(src[0][0])) {
+            fill = Import.grad(src[0]);
+        }
+        fill = Import.pattern(src[0]);
+    }
+    return Brush.stroke(fill, // paint
```
Comment (shamansir) [inline src/import/animatron-importer.js]: In removed lines above, first argument for `Brush.stroke` was `src[1]` for both array-case and non-array-case. Why was it changed to `src[0]` here, while `src[0]` is a width, as the line below states?

File: src/anm/graphics/brush.js (hunk 1 above)
Comment (mcm69) [inline src/anm/graphics/brush.js]: it could be in the previous version, thanks for noticing!

File: src/import/animatron-importer.js (hunk 2 above)
Comment (mcm69) [inline src/import/animatron-importer.js]: ¯\_(ツ)_/¯

Comment (shamansir) [inline src/import/animatron-importer.js]: I mean, may be export format was changed or it was a bug here before, but then the line below looks improper.

Comment (mcm69) [inline src/import/animatron-importer.js]: I mean "my bad"  ¯\_(ツ)_/¯

Comment (shamansir) [inline src/import/animatron-importer.js]: Uh, ok then :)

**Comment by shamansir** [comment]
lgtm now.

**Pull request merged**